### PR TITLE
feat: disable language/translations selection

### LINKED ISSUE #14527: Disable language selector, set `English` as default language and add option in advanced settings to manually enable its visualization
### Description
1. By default, the language selector MUST be disabled and the default language set is `English`. The design for this use case is below in comments

2. If an existing user already has a language different than `English`, this will be reset to `English`.
3. The user shall be able to enable the `language selector` by selecting the following option in `Advanced settings`:
`Enable language selector` --> New advanced option shall be added

    And then, the design should be the following one (existing one):

Screenshot 2024-04-26 at 10 45 10

4. Whenever a user disables the language selector, `English` must be set up as the default language.

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- make it conditionally available for testing via Settings/Advanced/Enable translations
- display a tooltip and infobox explaining the current and future situation
- upon app start, force the language to "en" if the translations are not enabled; and suggest app restart when they get disabled

Fixes #14527

### Affected areas

Settings/[Language,Advanced]

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Defaults:
image

Advanced toggle:
image

Translations enabled:
image

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Controls/StatusPickerButton.qml [hunk 1]
```diff
@@ -5,71 +5,65 @@ import QtQuick.Layouts 1.15
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
 import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
 
-Button {
+StatusButton {
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Controls/StatusPickerButton.qml:10]: The main purpose here is to allow having a tooltip over a disabled (non-interactive) button

File: ui/StatusQ/src/StatusQ/Controls/StatusPickerButton.qml [hunk 2]
```diff
@@ -5,71 +5,65 @@ import QtQuick.Layouts 1.15
...
     */
-    property StatusAssetSettings asset: StatusAssetSettings {
+    asset {
         width: 20
         height: 20
         imgIsIdenticon: false
     }
 
-    enum Type {
+    enum PickerType {
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Controls/StatusPickerButton.qml:27]: Fixing `enum` name clash

File: ui/app/AppLayouts/Profile/views/LanguageView.qml [hunk 3]
```diff
@@ -93,6 +95,16 @@ SettingsContentBase {
             Item { Layout.fillWidth: true }
             StatusListPicker {
                 id: languagePicker
+
+                button.interactive: root.languageSelectionEnabled
+                StatusToolTip {
+                    y: parent.height + Style.current.padding
+                    margins: 0
+                    visible: !root.languageSelectionEnabled && languagePicker.button.hovered
+                    orientation: StatusToolTip.Orientation.Bottom
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:104]: Can't put the tooltip below... the `z` order is totally messed up here due to bad `StatusListPicker` design

File: ui/app/AppLayouts/Profile/views/LanguageView.qml [hunk 4]
```diff
@@ -143,13 +155,28 @@ SettingsContentBase {
...
+            borderColor: Theme.palette.baseColor2
+            textColor: Theme.palette.directColor1
+            icon: "group-chat"
+            iconColor: Theme.palette.baseColor1
+            text: qsTr("We need your help to translate Status, so that together we can bring privacy and free speech to the people everywhere, including those who need it most.")
+            extraContentComponent: StatusFlatButton {
+                icon.name: "external-link"
+                text: qsTr("Learn more")
+                size: StatusBaseButton.Size.Small
+                onClicked: Global.openLink("https://status.app/translations")
```
Comment (alexjba) [inline ui/app/AppLayouts/Profile/views/LanguageView.qml:176]:
```suggestion
                onClicked: Global.openLinkWithConfirmation(Constants.externalStatusLinkWithHttps + '/translations',  Constants.externalStatusLinkWithHttps)
```

**Review by alexjba — APPROVED** [review]
LGTM! Minor comment on the `Learn more` link

Tested

File: ui/app/AppLayouts/Profile/views/LanguageView.qml (hunk 4 above)
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/LanguageView.qml]: Fixed!

**Review by jrainville — APPROVED** [review]

**Pull request merged**